src/dbnode/storage/index/aggregate_results.go

```diff
@@ -107,16 +107,70 @@ func (r *aggregatedResults) AddDocuments(batch []doc.Document) (int, error) {
 func (r *aggregatedResults) addDocumentsBatchWithLock(
 	batch []doc.Document,
 ) error {
-	for i := range batch {
-		err := r.addDocumentWithLock(batch[i])
-		if err != nil {
-			return err
+	for _, doc := range batch {
+		if r.aggregateOpts.Type == AggregateTagNamesAndValues {
```

Comment:
Maybe switch here and return if unknown?

Reply:
Good call :+1: